src/addrman.cpp

```diff
@@ -246,12 +246,12 @@ void AddrManImpl::Unserialize(Stream& s_)
 
     uint8_t compat;
     s >> compat;
-    const uint8_t lowest_compatible = compat - INCOMPATIBILITY_BASE;
-    if (lowest_compatible > FILE_FORMAT) {
+    const int lowest_compatible{int{compat} - int{INCOMPATIBILITY_BASE}};
```

Comment:
I think it should be like this:

```cpp
if (compat < INCOMPATIBILITY_BASE) {
    throw std::ios_base::failure(strprintf("Corrupted peers.dat: the compat value is lower than the expected 32"));
}
// original code follows that is now guaranteed not to underflow
const uint8_t lowest_compatible = compat - INCOMPATIBILITY_BASE;
...
```

The proposed change in this PR is ok in a sense that it will fix the underflow if too low value is read from disk. However, if that happens, then the printed message would be puzzling and we have to adjust that too. Normally, the meaning of the message is this:

> it is compatible with >=7, but I can only read up to 5 (so it is too high for me because 7>5)

and with a bricked `peers.dat` it could be:

> it is compatible with >=-10, but I can only read up to 5 (so it is too high for me because -10>5)

it is puzzling because -10 is not higher than 5.

Reply:
Thx, done